train.py

```diff
@@ -0,0 +1,302 @@
+        )
+        .decode("utf-8")
+        .split("\n")
+    )
+
+    negative_files = list(
+        set(
+            negative_keyword_examples
+            + away_negative_keyword_examples
+            + google_dataset
+            + mswc
+        )
+    )
+    negative_files.remove("")
+    negative_files = [f for f in negative_files if ".wav" in f]
+    print(f"Founded {len(negative_files)} negative files")
+
+    # Convert the obtained URIs to local path
+    external_dataset_path = data_dir / "external"
+    external_dataset_path_str = str(external_dataset_path)
+
+    negative_files = [
+        f.replace(
+            "gs://autonomy-vision/audios/external-datasets",
+            external_dataset_path_str,
+        )
+        for f in negative_files
+    ]
+
+    # Convert files to Path
+    positive_files = [Path(f) for f in positive_files]
+    negative_files = [Path(f) for f in negative_files]
+
+    if toy:
+        print(" --- Using toy dataset ---")
+        positive_files = positive_files[:200]
+        negative_files = negative_files[:2000]
+
+    model_dir = file_path.parent.parent.parent / "models" / "kwd" / "embedding_model"
+    # base_model = tf.keras.models.load_model(model_dir)
+    # embedding = tf.keras.models.Model(
+    #     name="embedding_model",
+    #     inputs=base_model.inputs,
+    #     outputs=base_model.get_layer(name="dense_2").output,
+    # )
+    # embedding.trainable = False
+
+    # Prepare data
+    negative_files_str = [str(path) for path in negative_files]
+
+    # Split the list of files into train, validation, and test sets
+    train_negatives, test_negatives = train_test_split(
+        negative_files_str, test_size=0.1, random_state=42
+    )
+
+    # Print the number of files in each set
+    print(f"Number of train negative files: {len(train_negatives)}")
+    print(f"Number of test negative files: {len(test_negatives)}")
+
+    # TODO: Change this split with one that takes into account the origin
+    # of the recordings (the bot, synthetic, etc.)
+    positive_files_str = [str(path) for path in positive_files]
+    # Split the list of files into train, validation, and test sets
+    train_samples, test_samples = train_test_split(
+        positive_files_str, test_size=0.2, random_state=42
+    )
+    train_samples, valid_samples = train_test_split(
+        train_samples, test_size=0.3, random_state=42
+    )
+
+    # Print the number of files in each set
+    print(f"Number of train files: {len(train_samples)}")
+    print(f"Number of validation files: {len(valid_samples)}")
+    print(f"Number of test files: {len(test_samples)}")
+
+    background_noise = data_dir / "ours" / "noise"
+
+    # Prepare the model
+    model_settings = standard_microspeech_model_settings(3)
+
+    if toy:
+        num_epochs = 1
+        num_batches = 1
+        backprop_into_embedding = False
+    else:
+        num_epochs = 10
+        num_batches = 2
+        backprop_into_embedding = True
+
```

Comment:
make these arguments of the main functon

Reply:
But, make sense taking into account that I'm also having "toy" as argument?

Reply:
Yes, toy override them

Reply:
aye aye

Reply:
Done, I left default the toy paramters and left a comment of recommended parameters with full training